src/pages/instances/forms/DuplicateInstanceSnapshotForm.tsx

```diff
@@ -0,0 +1,318 @@
+  instance: LxdInstance;
+  snapshot: LxdInstanceSnapshot;
+  close: () => void;
+}
+
+export interface LxdInstanceSnapshotDuplicate {
+  instanceName: string;
+  targetProject: string;
+  targetClusterMember?: string;
+  targetStoragePool: string;
+  stateful: boolean;
+}
+
+export const duplicateSnapshotPayload = (
+  values: LxdInstanceSnapshotDuplicate,
+  instance: LxdInstance,
+  snapshot: LxdInstanceSnapshot,
+) => {
+  const basePayload: Record<string, unknown> = {
+    name: values.instanceName,
+    description: instance.description,
+    architecture: instance.architecture,
+    ephemeral: snapshot.ephemeral,
+    mode: "pull",
+    devices: {
+      ...instance.devices,
+      root: {
+        path: "/",
+        type: "disk",
+        pool: values.targetStoragePool,
+      },
+    },
+  };
+
+  const source: Record<string, unknown> = {
+    project: instance.project,
+    type: "copy",
+    source: `${instance.name}/${snapshot.name}`,
+    base_image: snapshot.config["volatile.base_image"] ?? "",
+  };
+
+  if (snapshot.stateful && values.stateful) {
+    basePayload["stateful"] = true;
+    source["live"] = false;
+  }
+
+  return {
+    ...basePayload,
+    source,
+  };
+};
+
+const DuplicateInstanceSnapshotForm: FC<Props> = ({
+  instance,
+  snapshot,
+  close,
+}) => {
+  const toastNotify = useToastNotification();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const controllerState = useState<AbortController | null>(null);
+  const navigate = useNavigate();
+  const eventQueue = useEventQueue();
+
+  const { data: projects = [], isLoading: projectsLoading } = useQuery({
+    queryKey: [queryKeys.projects],
+    queryFn: fetchProjects,
+  });
+
+  const { data: storagePools = [], isLoading: storagePoolsLoading } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(instance.project),
+  });
+
+  const { data: clusterMembers = [], isLoading: clusterMembersLoading } =
+    useQuery({
+      queryKey: [queryKeys.cluster],
+      queryFn: fetchClusterMembers,
+      enabled: isClustered,
+    });
+
+  const { data: instances = [] } = useQuery({
+    queryKey: [queryKeys.instances],
+    queryFn: () => fetchInstances(instance.project),
+  });
+
+  const notifySuccess = (instanceName: string, instanceProject: string) => {
+    const message = (
+      <>
+        Created Instance <strong>{instanceName}</strong>.
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () =>
+          navigate(
+            `/ui/project/${instanceProject}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const getDuplicatedInstanceName = (instance: LxdInstance): string => {
+    const instanceNames = instances.map((instance) => instance.name);
+    const newInstanceName = truncateInstanceName(
+      instance.name,
+      "-snapshot-duplicate",
```

Comment:
```suggestion
      "-snapshot-copy",
```
Maybe better -- or we could also include the snapshot name here. so it is "instance-snapshot-copy"

Reply:
having the snapshot name is a great idea :+1: